projects/alexlab/index.js

```diff
@@ -1,44 +1,59 @@
-const { get } = require('../helper/http')
-
-// https://explorer.stacks.co/txid/SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.alex-vault?chain=mainnet
-// https://stacks-node-api.blockstack.org/extended/v1/address/SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.alex-vault/balances
-const ALEX_API = "https://api.alexlab.co/v1";
-
-async function fetch() {
-  const url = `${ALEX_API}/pool_token_stats`;
-  const alexStatsResponse = await get(url)
-
-  const valueLockedMap = {};
-  let totalValueLocked = 0;
-  for (const pool of alexStatsResponse) {
-    let poolValue = 0;
-    const poolToken = pool.pool_token;
-
-    if (poolToken == "age000-governance-token") {
-      poolValue = pool.price * pool.reserved_balance;
-    } else {
-      poolValue = pool.price * pool.total_supply;
-    }
-    totalValueLocked += poolValue;
-    valueLockedMap[poolToken] = poolValue;
-  }
-
-  return { tether: totalValueLocked };
-}
-
-async function staking() {
-  const url = `${ALEX_API}/stats/tvl`;
-  const alexResponse = await get(url)
-  return { tether: alexResponse.reserve_pool_value };
-}
-
-// node test.js projects/alexlab/index.js
+const axios = require("axios");
+const ADDRESSES = require("../helper/coreAssets.json");
+const { toUSDT } = require("../helper/balances");
+
+const pool_api = "https://api.alexlab.co/v2/public/pools";
+const price_api = "https://api.alexlab.co/v1/price/";
+const decimals = 1e18;
+
+const ownerTokens = [
+  "SP102V8P0F7JX67ARQ77WEA3D3CFB5XW39REDT0AM.token-alex",
+  "SP3K8BC0PPEVCV7NZ6QSRWPQ2JE9E5B6N3PA0KBR9.auto-alex-v2",
+];
+
+const getPools = async (url) => {
+  const { data } = await axios.get(url);
+  return data.data.map((pool) => ({
+    token_x: pool.token_x,
+    balance_x: pool.balance_x,
+    token_y: pool.token_y,
+    balance_y: pool.balance_y,
+  }));
+};
+
+const poolPriceInUSD = async (api, pools) => {
+  await Promise.all(
+    pools.map(async (pool) => {
+      const [{ data: priceX }, { data: priceY }] = await Promise.all([
```

Comment:
this would fetch price of same token multiple times, also relying on external service for price is a bad idea

check out `transformDexBalances` (https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/helper/portedTokens.js#L131)

this method is handing in computing tvl of uni v2 like dexes, if we have price of just one or two tokens in the dex, it can be used to price the rest, known tokens are auto pulled from here: https://github.com/DefiLlama/DefiLlama-Adapters/blob/main/projects/helper/coreAssets.json